slugid.js

```diff
@@ -42,13 +46,16 @@ exports.decode = function(slug) {
   return uuid.unparse(new Buffer(base64, 'base64'));
 };
 
-/** Generate a v4 (random) uuid and encode it to a slug */
+/** Returns a randomly generated uuid v4 complaint slug guaranteed not to have
+ * a leading `-` character */
 exports.v4 = function() {
-  var bytes   = uuid.v4(null, new Buffer(16));
-  var base64  = bytes.toString('base64');
+  do {
+    var bytes   = uuid.v4(null, new Buffer(16));
+    var base64  = bytes.toString('base64');
+  } while (/^\+/.test(base64));  // disallow leading '-' ('+' => '-' below)
```

Comment:
> @petemoore, okay... let's reduce complexity down to:
> ```js
> if (base64[0] === '+') {
>   base64 = 'A' + base64.substr(1);
> }
> ```
> 
> It's sufficiently random and solid...

Your proposed solution introduces a skew in the distribution of slugids. Twice as many of them start with an "A" as any other letter. It also unnecessarily reduces entropy. It is not clear to me what the benefit here would be. The code is not simpler, and the performance is not improved, yet comes at the cost of the skewed distribution and loss in entropy. Can you clarify what the issue is you see with my loop implementation? We established it is not performance or readability, so I am not sure what it is.